paasta_tools/check_marathon_services_replication.py

```diff
@@ -361,12 +361,24 @@ def get_smartstack_replication_for_attribute(attribute, service, namespace, blac
               (the dictionary will contain keys for unique all attribute values)
     """
     replication_info = {}
-    unique_values = mesos_tools.get_mesos_slaves_grouped_by_attribute(attribute=attribute, blacklist=blacklist)
+    filtered_slaves = mesos_tools.get_all_slaves_for_blacklist_whitelist(
+        blacklist=blacklist,
+        whitelist=[],
+    )
+    if not filtered_slaves:
+        raise mesos_tools.NoSlavesAvailableError
+
+    attribute_slave_dict = mesos_tools.get_mesos_slaves_grouped_by_attribute(
+        slaves=filtered_slaves,
+        attribute=attribute
+    )
```

Comment:
I guess you could move the grouping into a wrapper function too as we repeat the `get_all_slaves_for_blacklist_whitelist` and `get_mesos_slaves_grouped_by_attribute` twice.

Not massively fussed though